recipes/cpp-subprocess/all/conanfile.py

```diff
@@ -0,0 +1,26 @@
+from conans import ConanFile, tools
+
+
+class CppSubprocess(ConanFile):
+    name = "cpp-subprocess"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/nextsilicon/cpp-subprocess#readme"
+    topics = ("subprocess", "os", "fork")
+    description = ("Subprocessing with modern C++, "
+                   "The only goal was to develop something that is as close as"
+                   "python subprocess module in dealing with processes.")
+    no_copy_source = True
+
+    _source_subfolder = 'cpp-subprocess'
+
+    def source(self):
+        tools.get(**self.conan_data['sources'][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def package(self):
+        self.copy("subprocess.hpp", dst="include/cpp-subprocess", src=self._source_subfolder)
```

Comment:
I changed it in the `2.0.2` release to follow the conan package name namespace, both in `CMakeLists.txt`, and in `README`.

Not entirely sure what do you have in mind, but I don't have any strong feeling about it.

I feel that namespacing by package name is correct, and that subprocess is too general to be used as namespace, and can collide.

Reply:
https://github.com/nextsilicon/cpp-subprocess/commit/4df7d676d39debc4c42881461055353b02fa80d6

Reply:
ping

Reply:
`<subprocess/subprocess.h>` or `<subprocess.h>` seems to be the recommended way to include this header as of upstream README and CMakeLists. I mean that whether it's a good convention or not is irrelevant, conan recipes should honor upstream conventions.

Reply:
I don't think this is correct. This is a header only library, which is on purpose composed of a single file the user is expected to vendor in his source code where convenient to him.

That's why there's no really a recommended way. That's how single file header only works. Same for sqlite almagation.

The upstream haven't changed for a year, and is unlikely to have any major changes, the correct approach IMHO is to ask upstream to change the `README` which I intend to do, along with other related changes.

At any rate, I'm not going to battle over this hill, and if you wish to change it to `subprocess`, let it be so.

What do you think?

Reply:
Here's a use case for example https://github.com/google/google-ctf/blob/master/2019/quals/sandbox-devmaster-8001/src/client_lib.h#L29

`#include "third_party::subprocess"`

Reply:
Not sure to understand, it is vendored in google-ctf, it's unrelated.

Reply:
I believe that vendoring is the intended use case for this library, it's not packaged in any place AFAIU, so the fact that the CMakeLists contains a certain prefix is of little importance.

I changed it in my fork, submitted PR (not sure they'll be merged, as there are a lot of ourstanding PRs, none merged for quite a time).

Using prefix of `subprocess` both doesn't imply to the reader of the source of the library, and is ambigious. I don't think keeping the name that was in the CMakeLists.txt, and I see no evidence it was ever used, would help us, and I see a couple of ways it'd be problematic and will be confused with other things in big projects.

Reply:
Maybe ping other reviewers to see what they think about your proposal.

/cc @madebr @ericLemanissier @SSE4 @prince-chrismc @toge @Croydon @uilianries

Reply:
So https://github.com/nextsilicon/cpp-subprocess#integration has the subfolder,
but https://github.com/arun11299/cpp-subprocess hasn't.
Hence your pr at https://github.com/arun11299/cpp-subprocess/pull/72

Current cci practice is to use the *nomenclature* of the upstream project.
So for arun11299, this means not using a subfolder.
You can create a forked recipe nextsilicon-cpp-subprocess with modifications/patches.

The arun11299/cpp-subprocess project hasn't seen any activity for more then a year though.
But I don't think this is a reason to declare it dead yet.

Reply:
It's not our place as package maintainers to change how a user interacts with a project... Ideally switching to Conan is seamless, so madebr's comment rings home.

↪️ We should respect the current expectation of consumers

> So for arun11299, this means not using a subfolder.

I would like to see both since the CMake installs into a subfolder https://github.com/arun11299/cpp-subprocess/blob/fa58db9d6e250168b864970051b07a962b67d915/CMakeLists.txt#L12 but the examples does use one. e.g copy in package to subfolder but also add the subfolder in the includes so consumers get the choice.

***

That being said since this is packaging the fork and not the main repository I think it acceptable to follow the "new" folder naming since this is a different project

For myself, this would require changing the name of the package to include the fork 👍

Reply:
@madebr @prince-chrismc my main point is that the current interface of users with the library, and also the intention of the author IMHO, is to vendor the hpp file into your project. There's no intention for you to install it and use the `subprocess/subprocess.hpp` as specified in the original project, and searching cs.github.com doesn't show anyone actually doing that.

So I don't think there _is_ any nomenclature other than vendoring it in your `third_party` folder.

Anyohow, if @prince-chrismc 's decision is finaly, I renamed the folder and kept the subfolder.

I hope the original author would merge back the fix to the prefix, and all will be settled.

Reply:
The C++ ecosystem has always been the wild west of "how to use my library" with Conan we hopefully can have something more sensible.

It hard to guess  the authors intentions and the best we can do is follow their docs and example 🤷 

> if @prince-chrismc 's decision is final

As much as i wish I was a real prince, that is merely my recommendation but we appreciate your openness to our ideas ❤️ 

Hopefully this is a good compromise